cpp/include/cudf/detail/aggregation/aggregation.hpp

```diff
@@ -109,13 +159,38 @@ struct lead_lag_aggregation final : derived_aggregation<lead_lag_aggregation> {
   size_t hash_impl() const { return std::hash<size_type>()(row_offset); }
 };
 
+/**
+ * @brief Derived class for specifying a mean aggregation
+ */
+struct mean_aggregation final : aggregation {
+  mean_aggregation() : aggregation{MEAN} {}
+
+  std::vector<aggregation::Kind> get_simple_aggregations(data_type col_type) const override
+  {
+    CUDF_EXPECTS(is_fixed_width(col_type), "MEAN aggregation expects fixed width type");
+    return {aggregation::SUM, aggregation::COUNT_VALID};
+  }
+  void finalize(aggregation_finalizer& finalizer) override { finalizer.visit(*this); }
+};
+
 /**
  * @brief Derived class for specifying a standard deviation/variance aggregation
  */
 struct std_var_aggregation final : derived_aggregation<std_var_aggregation> {
-  std_var_aggregation(aggregation::Kind k, size_type ddof) : derived_aggregation{k}, _ddof{ddof} {}
+  std_var_aggregation(aggregation::Kind k, size_type ddof) : derived_aggregation{k}, _ddof{ddof}
+  {
+    CUDF_EXPECTS(k == aggregation::STD or k == aggregation::VARIANCE,
+                 "std_var_aggregation can accept only STD, VARIANCE");
```

Comment:
I don't understand why this class is allowed to accept an `arregation::Kind` argument. Are you trying to use the same `aggregation` class for two different aggregations, i.e., `STD` and `VAR`?

Reply:
Yes. STD, VAR both use `std_var_aggregation`.
My guess is that STD, VAR both uses `ddof` argument, so they used a single class and denoted which aggregation by `arregation::kind`.

Reply:
They should have their own concrete `aggregation` derived types. It's too confusing to have several aggs share the same type. It should be one type per agg.

Reply:
Right. Ideally, each aggregation should be its own type. But we use enum. so whenever some functionality is common among aggregations, they are combined to a single type. 
For example, LEAD, LAG has single type `lead_lag_aggregation`.

Reply:
>  so whenever some functionality is common among aggregations, they are combined to a single type.

This is precisely the problem. Each aggregation should have it's own type. If there is significant common functionality between two or more aggregations, they can inherit from a common base.